src/coreclr/tools/Common/TypeSystem/IL/Stubs/AsyncResumptionStub.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+using Internal.IL;
+using Internal.IL.Stubs;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler
+{
+    public partial class AsyncResumptionStub : ILStubMethod
+    {
+        private readonly MethodDesc _owningMethod;
+        private MethodSignature _signature;
+
+        public AsyncResumptionStub(MethodDesc owningMethod)
+        {
+            Debug.Assert(owningMethod.IsAsyncVariant()
+                || (owningMethod.IsAsync && !owningMethod.Signature.ReturnsTaskOrValueTask()));
```

Comment:
One more place to use the extension method mentioned in the other PR

Reply:
Yep. Is it necessary to block this PR until that one merges?